lua/efmls-configs/formatters/smlfmt.lua

```diff
@@ -0,0 +1,13 @@
+-- Metadata
+-- languages: sml
+-- url: https://github.com/shwestrick/smlfmt
+
+local fs = require('efmls-configs.fs')
+
+local formatter = 'smlfmt'
+local command = string.format("%s --preview-only ${FILENAME} | sed '1d;$d'", fs.executable(formatter))
+
+return {
+  formatCommand = command,
+  formatStdin = true,
```

Comment:
Set this to `false` since we don't want changes from stdin.

Reply:
I'm not too familiar with efm language server, but both these changes don't seem to fix the issue.

```
local fs = require("efmls-configs.fs")

local formatter = "smlfmt"
local command = string.format("%s --force ${INPUT}", fs.executable(formatter))

return {
  formatCommand = command,
  formatStdin = false,
}
```

With these changes, this how formatting happens: [example](https://asciinema.org/connect/31cc47ff-616b-4e83-8fa9-ea7bf1ff7cce)

I've been using [my fork of smlfmt](https://github.com/j-barnak/smlfmt) so that this would work,

```
local fs = require("efmls-configs.fs")

local formatter = "smlfmt"
local command = string.format("%s --preview-only ${INPUT}", fs.executable(formatter))

return {
  formatCommand = command,
  formatStdin = true,
}
```

But, honestly, I  wouldn't like to use my fork.

Reply:
Okay I downloaded `mlton` and compiled `smlfmt` on my machine and went thru the docs, and it looks like its simpler than it seems.

No need to add any flags, just provide the tool and set `formatStdin` back to `true`:

```lua
return {
  formatCommand = fs.executable(formatter),
  formatStdin = true,
}
```

Please give that a try and let me know if it works on your side.

Reply:
This also doesn't work: [video](https://asciinema.org/a/ijgNxKpfLQTMUdDvyL4Eo58qM)

```
local fs = require("efmls-configs.fs")

local formatter = "/home/jared/Installs/smlfmt/smlfmt" -- non-forked smlfmt
local command = string.format("%s  ${INPUT}", fs.executable(formatter))

return {
  formatCommand = command,
  formatStdin = true,
```

Reply:
You don't pass the `${INPUT}` its just the tool. Hence, its just `formatCommand = fs.formatter(formatter)`.

Reply:
Oh yeah! That works!

Reply:
Nice! Ready to merge whenever you've applied the changes :)

Reply:
Thank so much! Just pushed the changes.